src/Mvc/src/Microsoft.AspNetCore.Mvc.Razor/Compilation/CompiledViewDescriptor.cs

```diff
@@ -80,5 +82,23 @@ public CompiledViewDescriptor(RazorCompiledItem item, RazorViewAttribute attribu
         /// Gets the type of the compiled item.
         /// </summary>
         public Type Type => Item?.Type ?? ViewAttribute?.ViewType;
+
+        internal Func<IRazorPage> CreatePageFactory()
```

Comment:
Is this really the best place for this? This class is just random junk. It has like 5 different consumers and they all do different things.

Reply:
It's also used in public extensibility.